File: dev/app_ios/StayOn/Services/Notification/NotificationTextBuilder.swift
import Foundation

enum NotificationTextBuilder {
    /// Current tone style, set from settings when user changes it.
    static var currentStyle: ToneStyle = .lightSupervision
    static func medicationTitle(style: ToneStyle = .lightSupervision) -> String {
        switch style {
        case .lightSupervision: return "该吃药了，别拖。"
        case .gentle: return "该吃药了，记得按时。"
        case .serious: return "请立即服药！"
        }
    }

    static func medicationBody(medicationName: String, dosage: String?) -> String {
        if let dosage, !dosage.isEmpty {
            return "\(medicationName) \(dosage)"
        }
        return medicationName
    }

    static func sleepPreReminderTitle(style: ToneStyle = .lightSupervision) -> String {
        switch style {
        case .lightSupervision: return "差不多该收了。"
        case .gentle: return "准备睡觉了。"
        case .serious: return "请注意休息时间。"
        }
    }

    static func sleepPreReminderBody(minutes: Int, style: ToneStyle = .lightSupervision) -> String {
        switch style {
        case .lightSupervision: return "\(minutes)分钟后准备睡觉，别又拖到太晚。"
        case .gentle: return "\(minutes)分钟后该休息了，早点睡。"
        case .serious: return "您将在\(minutes)分钟后到达目标就寝时间。"
        }
    }

    static func sleepReminderTitle(style: ToneStyle = .lightSupervision) -> String {
        switch style {
        case .lightSupervision: return "到点了，该睡了。"
        case .gentle: return "到睡觉时间了。"
        case .serious: return "就寝时间已到。"
        }
    }

    static func sleepReminderBody(style: ToneStyle = .lightSupervision) -> String {
        switch style {
        case .lightSupervision: return "今晚别乱来，按时收工。"
        case .gentle: return "好好休息，明天会更好。"
        case .serious: return "睡眠不足影响健康，请立即休息。"
        }
    }

    static func mealReminderTitle(style: ToneStyle = .lightSupervision) -> String {
        switch style {
        case .lightSupervision: return "吃之前先看一眼。"
        case .gentle: return "吃饭前注意一下。"
        case .serious: return "饮食提醒。"
        }
    }

    static func mealReminderBody(tags: [String], customNote: String?) -> String {
        var parts: [String] = tags
        if let note = customNote, !note.isEmpty {
            parts.append(note)
        }
        return parts.joined(separator: "，")
    }

    static func nightSnackTitle(style: ToneStyle = .lightSupervision) -> String {
        switch style {
        case .lightSupervision: return "今晚夜宵就算了。"
        case .gentle: return "晚上少吃点。"
        case .serious: return "请控制夜宵摄入。"
        }
    }

    static func nightSnackBody(style: ToneStyle = .lightSupervision) -> String {
        switch style {
        case .lightSupervision: return "现在忍住，明天会感谢你。"
        case .gentle: return "早点休息，对胃好。"
        case .serious: return "夜宵会增加身体负担。"
        }
    }
}

File: dev/app_android/app/src/main/java/com/stayon/app/domain/util/NotificationTextBuilder.kt
package com.stayon.app.domain.util

import com.stayon.app.domain.enums.ToneStyle

object NotificationTextBuilder {

    var currentStyle: ToneStyle = ToneStyle.LIGHT_SUPERVISION

    fun medicationTitle(): String = when (currentStyle) {
        ToneStyle.LIGHT_SUPERVISION -> "该吃药了，别拖。"
        ToneStyle.GENTLE -> "该吃药了，记得按时。"
        ToneStyle.SERIOUS -> "请立即服药！"
    }

    fun medicationBody(name: String, dosage: String?): String =
        if (!dosage.isNullOrEmpty()) "$name $dosage" else name

    fun sleepPreTitle(): String = when (currentStyle) {
        ToneStyle.LIGHT_SUPERVISION -> "差不多该收了。"
        ToneStyle.GENTLE -> "准备睡觉了。"
        ToneStyle.SERIOUS -> "请注意休息时间。"
    }

    fun sleepPreBody(minutes: Int): String = when (currentStyle) {
        ToneStyle.LIGHT_SUPERVISION -> "${minutes}分钟后准备睡觉，别又拖到太晚。"
        ToneStyle.GENTLE -> "${minutes}分钟后该休息了，早点睡。"
        ToneStyle.SERIOUS -> "您将在${minutes}分钟后到达目标就寝时间。"
    }

    fun sleepTitle(): String = when (currentStyle) {
        ToneStyle.LIGHT_SUPERVISION -> "到点了，该睡了。"
        ToneStyle.GENTLE -> "到睡觉时间了。"
        ToneStyle.SERIOUS -> "就寝时间已到。"
    }

    fun sleepBody(): String = when (currentStyle) {
        ToneStyle.LIGHT_SUPERVISION -> "今晚别乱来，按时收工。"
        ToneStyle.GENTLE -> "好好休息，明天会更好。"
        ToneStyle.SERIOUS -> "睡眠不足影响健康，请立即休息。"
    }

    fun mealTitle(): String = when (currentStyle) {
        ToneStyle.LIGHT_SUPERVISION -> "吃之前先看一眼。"
        ToneStyle.GENTLE -> "吃饭前注意一下。"
        ToneStyle.SERIOUS -> "饮食提醒。"
    }

    fun mealBody(tags: List<String>, note: String?): String =
        (tags + if (!note.isNullOrEmpty()) listOf(note) else emptyList()).joinToString("，")

    fun nightSnackTitle(): String = when (currentStyle) {
        ToneStyle.LIGHT_SUPERVISION -> "今晚夜宵就算了。"
        ToneStyle.GENTLE -> "晚上少吃点。"
        ToneStyle.SERIOUS -> "请控制夜宵摄入。"
    }

    fun nightSnackBody(): String = when (currentStyle) {
        ToneStyle.LIGHT_SUPERVISION -> "现在忍住，明天会感谢你。"
        ToneStyle.GENTLE -> "早点休息，对胃好。"
        ToneStyle.SERIOUS -> "夜宵会增加身体负担。"
    }
}
